include/crypto/algapi.h

```diff
@@ -180,6 +181,23 @@ static inline void crypto_xor_cpy(u8 *dst, const u8 *src1, const u8 *src2,
 	}
 }
 
+static inline void crypto_xor_reverse(u8 *dst, const u8 *src, unsigned int size)
```

Comment:
Why is `crypto_xor_reverse()` needed?  If this is needed, it should be added in a separate patch.

If this is just an optimization to avoid copying each block of data to a stack buffer, I don't think you should worry about that too much.  This is just the generic implementation, and people will use one the architecture-optimized implementation if they care about performance.